### PR TITLE
Update sethBarberee Userspace

### PR DESCRIPTION
**Pull request by SethBarberee**
## Description
* Did a large overhaul on my userspace to account for my shift from Iris to Sinc.
* Unified all my RGB/Tap Dance work into my userspace
* Doing a little prep in my userspace for a crkbd that I keep putting off.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Updates #12160 for my Iris keymap

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by SethBarberee** [comment]
Errors out bc of `keebio/sinc/rev1/config.h` using the `RGBLIGHT_MODE_RAINBOW_SWIRL` define. I turn off `RGBLIGHT_ANIMATIONS` so causes the error.  `RGBLIGHT_DEFAULT_MODE` isn't used anymore since #12138. Would it be fine if I attached a patch to my PR fixing that for both Rev 1 and 2 Sinc?

File: keyboards/keebio/iris/keymaps/sethBarberee/config.h [hunk 1]
```diff
@@ -30,32 +30,13 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
+#undef RGBLED_SPLIT
+#define RGBLED_SPLIT { RGBLED_NUM, 0}
 
 #if !defined(NO_DEBUG) && !defined(CONSOLE_ENABLE)
 #define NO_DEBUG
 #endif // !NO_DEBUG
 #if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
 #define NO_PRINT
 #endif // !NO_PRINT
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/sethBarberee/config.h:42]: These aren't needed, either.

File: keyboards/keebio/sinc/keymaps/sethBarberee/config.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//#       define NORMAL_MODE RGBLIGHT_MODE_TWINKLE + 4
+//#   endif
+//#endif
+
+#if !defined(NO_DEBUG) && !defined(CONSOLE_ENABLE)
+#define NO_DEBUG
+#endif // !NO_DEBUG
+#if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
+#define NO_PRINT
+#endif // !NO_PRINT
```
Comment (drashna) [inline keyboards/keebio/sinc/keymaps/sethBarberee/config.h:37]:
```suggestion
```

File: keyboards/keebio/sinc/keymaps/sethBarberee/keymap.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keebio/sinc/keymaps/sethBarberee/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15


Also, if you move `#include QMK_KEYBOARD_H` to your sethBarberee.h file, you'd only need the one include.

File: users/sethBarberee/config.h [hunk 4]
```diff
@@ -1,8 +1,54 @@
+#pragma once
```
Comment (drashna) [inline users/sethBarberee/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/rgb_light.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#include "rgb_light.h"
```
Comment (drashna) [inline users/sethBarberee/rgb_light.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/rgb_light.h [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline users/sethBarberee/rgb_light.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/sethBarberee.c [hunk 7]
```diff
@@ -1,45 +1,70 @@
 #include "sethBarberee.h"
```
Comment (drashna) [inline users/sethBarberee/sethBarberee.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/sethBarberee.h [hunk 8]
```diff
@@ -1,20 +1,35 @@
-#ifndef USERSPACE
-#define USERSPACE
+#pragma once
```
Comment (drashna) [inline users/sethBarberee/sethBarberee.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/tap_dance.c [hunk 9]
```diff
@@ -0,0 +1,74 @@
+#include "tap_dance.h"
```
Comment (drashna) [inline users/sethBarberee/tap_dance.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/tap_dance.h [hunk 10]
```diff
@@ -0,0 +1,28 @@
+#pragma once
```
Comment (drashna) [inline users/sethBarberee/tap_dance.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/sethBarberee/wrappers.h [hunk 11]
```diff
@@ -0,0 +1,40 @@
+#pragma once
```
Comment (drashna) [inline users/sethBarberee/wrappers.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by SethBarberee** [comment]
Updated and addressed comments. Would appreciate opinion on the sinc build issue and whether I can include the patch for that build issue.

**Review by skullydazed — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]
Not *strictly* necessary, but it is recommended to lowercase your keymap & userspace names.

**Pull request merged**